R/shinyShortcut.R

```diff
@@ -0,0 +1,110 @@
+#' Create Shiny App Shortcut
+#'
+#' Creates an executable file that, when ran, will open the co-located shiny
+#' app using the user's default internet browser.
+#'
+#' @param shinyDirectory The top-level directory containing shiny code and
+#' materials. Also where the final shortcut file will be saved.
+#' @param OS The operating system that R is being ran on.
+#'
+#' @details Calling \code{shinyShortcut} will write an executable file
+#' that when called will run the shiny app in the default browser.
+#' This executable is stored in a created folder named  .shiny_run'
+#' whilst a shortcut that will silently run the executable
+#' is saved in the home directory as 'shinyShortcut' (this will
+#' be a .vbs script on windows or a .desktop file on unix systems).
+#'
+#' @examples
+#' shinyShortcut()
+#' @export
+shinyShortcut <- function(shinyDirectory = getwd(), OS = .Platform$OS.type) {
+
+  # if relevant files exist delete them first
+  unlink(".shiny_run", recursive = TRUE, force = TRUE)
+  unlink("shiny_run.desktop", recursive = FALSE, force = TRUE)
+  unlink("shinyShortcut.vbs", recursive = FALSE, force = TRUE)
+  unlink("shinyShortcut.cmd", recursive = FALSE, force = TRUE)
+
+  dir.create(paste(shinyDirectory, ".shiny_run", sep = "/"))
+
+  if (OS == "windows") {
+
+    # write batch file to .shiny_run
+    rscriptForwardDash <-
+      paste(R.home(), "bin/Rscript.exe", sep = "/")
+    rscript <- gsub("/", "\\\\", rscriptForwardDash)
+
+    shinyCommand <- paste0(
+      "shiny::runApp('", shinyDirectory, "',",
```

Comment:
Should be escaped--in particular, `'` characters need to be replaced with `\\'`.